torch/utils/_sympy/value_ranges.py

```diff
@@ -9,10 +10,31 @@
 
 log = logging.getLogger(__name__)
 
+SympyBoolean = sympy.logic.boolalg.Boolean
+
+def assert_simple_sympy(e):
+    if isinstance(e, sympy.Expr):
+        assert not e.free_symbols, f"has free variables {ee}"
+        assert e != sympy.nan
+    elif isinstance(e, BooleanAtom):
+        pass
+    else:
+        raise AssertionError(f"not simple sympy type {type(e)}")
```

Comment:
We will eventually have free symbols when we run this on dynamic shapes to optimize indexing to int32, since we will be creating expressions from the inputs. But this is fine for now.

Reply:
have comments about this now